File: src/2024/day-01/typescript/part-1.ts
export function Part1(input: string) {
  return input
    .trim()
    .split("\n")
    .map((line) => Array.from(line.trim().matchAll(/\d/g)).map(Number))
    .map((numbers) => numbers[0] * 10 + numbers[numbers.length - 1])
    .reduce((a, b) => a + b);
}

File: src/2024/day-01/rust/src/part_1.rs
pub fn part_1(input: &str) -> i32 {
  input
    .trim()
    .lines()
    .map(|ln| ln.trim().chars().filter_map(|c| c.to_digit(10)).collect::<Vec<u32>>())
    .map(|digits| {
      let first = digits.first().unwrap();
      let last = digits.last().unwrap();
      (*first as i32) * 10 + (*last as i32)
    })
    .sum()
}
